samcli/commands/init/interactive_init_flow.py

```diff
@@ -183,25 +183,33 @@ def _get_runtime_from_image(image):
 
 
 def _get_dependency_manager(dependency_manager, runtime):
-    if not dependency_manager:
-        valid_dep_managers = RUNTIME_TO_DEPENDENCY_MANAGERS.get(runtime)
-        if valid_dep_managers is None:
-            dependency_manager = None
-        elif len(valid_dep_managers) == 1:
-            dependency_manager = valid_dep_managers[0]
-        else:
-            choices = list(map(str, range(1, len(valid_dep_managers) + 1)))
-            choice_num = 1
-            click.echo("\nWhich dependency manager would you like to use?")
-            for dm in valid_dep_managers:
-                msg = "\t" + str(choice_num) + " - " + dm
-                click.echo(msg)
-                choice_num = choice_num + 1
-            choice = click.prompt("Dependency manager", type=click.Choice(choices), show_choices=False)
-            dependency_manager = valid_dep_managers[int(choice) - 1]  # zero index
+    valid_dep_managers = RUNTIME_TO_DEPENDENCY_MANAGERS.get(runtime)
+    if dependency_manager:
+        _validate_runtime_dependancy_manager(dependency_manager, runtime)
```

Comment:
I think you can simply put the lines 208 -> 210 here instead, this will also avoid repeating the call to `RUNTIME_TO_DEPENDENCY_MANAGERS.get(runtime)`